src/Http/Http.Results/src/AcceptedAtRouteHttpResult.cs

```diff
@@ -44,13 +55,21 @@ public AcceptedAtRouteResult(object? routeValues, object? value)
     /// <summary>
     /// Gets the route data to use for generating the URL.
     /// </summary>
-    public RouteValueDictionary RouteValues { get; }
+    public RouteValueDictionary? RouteValues { get; }
+
+    /// <summary>
+    /// Gets the HTTP status code.
+    /// </summary>
+    public int StatusCode => StatusCodes.Status202Accepted;
+
+    /// <inheritdoc/>
+    public Task ExecuteAsync(HttpContext httpContext)
+        => HttpResultsWriter.WriteResultAsJsonAsync(httpContext, Value, StatusCode, configureResponseHeader: ConfigureResponseHeaders);
 
-    /// <inheritdoc />
-    protected override void ConfigureResponseHeaders(HttpContext context)
+    private void ConfigureResponseHeaders(HttpContext context)
     {
         var linkGenerator = context.RequestServices.GetRequiredService<LinkGenerator>();
-        var url = linkGenerator.GetUriByAddress(
+        var url = linkGenerator.GetUriByRouteValues(
```

Comment:
Does this have a functional impact?

Reply:
I don't think it does have impact. If I understood the biggest difference is a RouteValueAddress object will be create by the new method that contains the AmbientValues and RouteValues. That said, I made this change when I added the interfaces, since the `GetUriByAddress` does not allow nullable RouteValues, but, since we removed I can rollback this change if that is preferable

Reply:
I reverted to the previous behavior, and we can discuss it outside of this PR